crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -635,6 +653,38 @@ impl RpcServerConfig {
 
         Ok(server)
     }
+
+    fn create_cors_layer(http_cors_domains: Option<String>) -> Result<Option<CorsLayer>, RpcError> {
+        let mut cors = None;
+        if let Some(domains) = http_cors_domains {
+            match domains.as_str() {
+                "*" => {
+                    cors = Some(
+                        CorsLayer::new()
+                            .allow_methods([Method::GET, Method::POST])
+                            .allow_origin(Any)
+                            .allow_headers(Any),
+                    );
+                }
+                "" => {}
+                _ => {
+                    let origins = domains
+                        .split(",")
+                        .map(|domain| domain.parse::<HeaderValue>())
+                        .collect::<Result<Vec<HeaderValue>, _>>();
```

Comment:
Nit I'd add a CoraDomains type that impls FromStr and includes this logic, and do Option<CorsDomains> in the upstream cli code